datajoint/schema.py

```diff
@@ -1,15 +1,59 @@
+from operator import itemgetter
 import pymysql
 import logging
 
-from . import conn, DataJointError, NoDefinitionError
+from . import conn, DataJointError
+from datajoint.utils import to_camel_case
 from .heading import Heading
 from .base_relation import BaseRelation
-from .user_relations import Part
+from .user_relations import Part, Computed, Imported, Manual, Lookup
 import inspect
+
 logger = logging.getLogger(__name__)
 from warnings import warn
 
 
+class SchemaFactory:
```

Comment:
I agree that there can be better names. But I think the name `create_relation_class_from_database` is too long and I am hesitant to name a class with lower case underscore style name. To the outside this suggests that it is a function and does therefore not have a state (which is not entirely true for python functions but expected anyway). I also like the name `Factory` because it is essentially a factory design pattern.  I'll think about something clearer.